clientapi/routing/register.go

```diff
@@ -945,6 +973,17 @@ func completeRegistration(
 	}
 	sessions.addCompletedRegistration(sessionID, result)
 
+	defer func() {
```

Comment:
I think we can remove the `defer` at this point (even though the defer makes the `post register behavior` clearer).
We should check that `accType` is not `AccountTypeAppService` and/or that `appserviceID` is empty, not sure how appservices react if they are automatically joined to rooms on registration.

Reply:
So what I need is to remove the defer func wrapper and add a check of `accType` and `appserviceID` ?
```
// POST register behavior: check if the user is a normal user.
// If the user is a normal user, add user to room specified in the configuration "auto_join_rooms".
if accType != userapi.AccountTypeAppService || appserviceID == "" {
	for room := range cfg.AutoJoinRooms {
        	...
	}
}
```

Reply:
Yep, thinking about it, it's probably better to use `&&`, but that looks fine otherwise.